### DISCUSSION TITLE
[Bug]: Detect/Record/Snapshot icons get in wrong state

### DISCUSSION THREAD

**Discussion by leccelecce** [blakeblackshear/frigate#14801] [Report a Bug]
### Checklist

- [X] I have updated to the latest available Frigate version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.
- [X] I have tried reproducing the issue in [incognito mode](https://www.computerworld.com/article/1719851/how-to-go-incognito-in-chrome-firefox-safari-and-edge.html) to rule out problems with any third party extensions or plugins I have installed.

### Describe the problem you are having

I'm noticing that after a restart of Frigate (via config save-and-restart), these icons frequently are in the wrong state.

![image](https://github.com/user-attachments/assets/0962ef5f-d4c0-4279-8b8f-d8ede56f36e3)

For example, if I've disabled detect, recording and snapshots via HASS, on restart the icons still show as disabled, but I know recording and snapshots will be back on.

I need to do a Ctrl+Shift+R refresh to fix them, which should not be necessary.

### Steps to reproduce

1. See above

### Version

0.15.0-1fc4af9

### Frigate config file

```yaml
n/a
```

### docker-compose file or Docker CLI command

```yaml
n/a
```

### Relevant Frigate log output

```shell
n/a
```

### Relevant go2rtc log output

```shell
n/a
```

### Operating system

Debian

### Install method

Docker Compose

### Network connection

Wired

### Camera make and model

n/a

### Screenshots of the Frigate UI's System metrics pages

n/a

**Comment by hawkeye217** [comment]
This is a duplicate of https://github.com/blakeblackshear/frigate/issues/11314 and will be improved in a future version.

**Reply by leccelecce** [reply]
I'm not sure that it is?

My issue is that Frigate is incorrectly reporting the state of detect/record/snapshot.

For example, it's showing that all three are inactive, but I can see recording in history.

**Reply by hawkeye217** [reply]
It's related, and it has to do with the way the backend passes around these states. The frontend won't receive those states automatically after a restart and a disconnect of the websocket. Reloading the page calls `onConnect` and fetches fresh values from the backend. 

Both issues will be fixed with the way we plan to refactor the backend.

**Reply by leccelecce** [reply]
Is the refactor likely in 0.15 now betas have started or is it likely a subsequent release? 

I may send a PR because I'm not comfortable with the silent failure - everything else in the UI works fine after a restart and there are no visible errors showing that the WS connection has been lost. It feels pretty dangerous to have a situation when you think you are or aren't recording/detecting based on the UI, and it's wrong.

Would you accept a PR for a visible warning in the UI e.g. in the status bar?

**Reply by hawkeye217** [reply]
We've known about this issue and Frigate has actually behaved like this for a while, and this is why the UI automatically reloads when restarting from the menu in the UI. When restarting from Docker, I always have to reload my browser as some streams don't automatically reconnect. 

The refactor won't likely happen for 0.15, but perhaps for 0.16.

I don't think any action is necessary at this point, but curious to see what the other maintainers think. @NickM-27

**Reply by leccelecce** [reply]
Ah, I think most of my restarts are from the Config->Save and Restart button, which doesn't trigger the popup and auto-reload countdown. 

How about changing the config page Save and Restart to trigger the same as the Restart Frigate option?

**Reply by hawkeye217** [reply]
Sure, that can be improved. https://github.com/blakeblackshear/frigate/pull/14815

**Reply by leccelecce** [reply]
Thanks!

**Reply by leccelecce** [reply]
I've realised another issue here. Restart triggered from desktop; a day later I realise my phone is showing the wrong state, and my kiosk tablet.

The problem is that most of the UI works between restarts just fine- moving between pages, viewing video - so it's not immediately obvious at all that you've forgotten and are seeing some stale data.

It's also not practical to go around multiple devices and refresh them all every restart. 

WS auto-reconnection would seem to be the most obvious solution I think?

**Reply by hawkeye217** [reply]
The websocket does automatically reconnect, and I can make a small improvement in there that will only partially (and unreliably) improve this.

The primary problem is with the dispatcher on the backend not reliably receiving the correct states because of timing issues on startup. This is what we intend to improve in future versions.

**Reply by hawkeye217** [reply]
https://github.com/blakeblackshear/frigate/pull/14847
